algebra-core/src/serialize/mod.rs

```diff
@@ -284,6 +321,22 @@ macro_rules! impl_prime_field_serializer {
             fn serialized_size(&self) -> usize {
                 Self::SERIALIZED_SIZE
             }
+
+            fn serialize_unchecked<W: crate::io::Write>(
+                &self,
+                writer: &mut W,
+            ) -> Result<(), crate::serialize::SerializationError> {
+                const BYTE_SIZE: usize = $byte_size;
+
+                let (_, output_byte_size) =
+                    crate::serialize::buffer_bit_byte_size($field::<P>::size_in_bits());
+
+                let mut bytes = [0u8; BYTE_SIZE];
+                self.write(&mut bytes[..])?;
+
+                writer.write_all(&bytes[..output_byte_size])?;
+                Ok(())
```

Comment:
This code is basically doing the same as the `serialize` or calling `self.serialize_with_flags(writer, crate::serialize::EmptyFlags)`.
The only difference is the handling of the empty flags.
Did you implement this for performance reasons?

Reply:
Yeah, I didn't want the additional overhead from the flag conversion. But I guess I should benchmark and compare the difference.

Reply:
There was no difference =D